### PR TITLE
Linux daemon issue

### LINKED ISSUE #2561: 🐛 [Bug] Vmangos has been increasing physical memory without an upper limit
## 🐛 Bug report
Vmangos has been continuously increasing the use of physical memory without limits
Even if I set it to run in the foreground or background, it will continue to increase the use of physical memory until the process crashes

The server is ubuntu22.04

### PR DESCRIPTION
**Pull request by imranh2**
## 🍰 Pullrequest

In an attempt to fix the memory leak & process going nuts when launched in `daemon` mode with the changes I made in https://github.com/vmangos/core/pull/2527, simply don't launch the CLI when running in this mode. Rationale is that you can't even use the CLI easily if launched like this.

There's also a change around paths and what your relative path is in `daemon` mode, making it *more consistent* with non daemon mode.

Also a small memory leak fix for the linux mangosd CLI.

### Proof

- None

### Issues

- fixes #2561 

### How2Test

- None

### Todo / Checklist

- [X] None

### CODE REVIEW

File: src/mangosd/CliRunnable.cpp [hunk 1]
```diff
@@ -144,6 +144,9 @@ void CliRunnable::operator()()
                 continue;
             }
             sWorld.QueueCliCommand(new CliCommandHolder(0, SEC_CONSOLE, nullptr, command.c_str(), &utf8print, &commandFinished));
+        #ifndef WIN32
+        free(command_str);
```
Comment (ratkosrb) [inline src/mangosd/CliRunnable.cpp:149]: the indentation does not match the rest of the function

**Comment by imranh2** [comment]
Rebased changes / fixed merge conflict.

File: src/mangosd/CliRunnable.cpp (hunk 1 above)
Comment (imranh2) [reply src/mangosd/CliRunnable.cpp:149]: Indenting aligns? It's intentionally after the if, so it's free'd even if it's nullptr which is ok to do.

Comment (mserajnik) [reply src/mangosd/CliRunnable.cpp:149]: > It's intentionally after the if

Looks to be inside the `if (command_str != nullptr)` to me though.

Comment (imranh2) [reply src/mangosd/CliRunnable.cpp:149]: 🤦 I'm clealy an idiot, thanks for the spotting this both!

**Review by imranh2 — COMMENTED** [review]
Rewrote commit to place the code in the right place

**Pull request merged**